File: benchmark/results/python-multilang/run-002/a-mcp/output/catalog/services/product_service.py
from typing import Any, Dict, List, Optional
from catalog.aggregates.product import Product

"""ProductService service."""
class ProductService:
    def create(self, input: Dict[str, Any]) -> Product:
        raise NotImplementedError('not implemented')
    def findById(self, id: str) -> Optional[Product]:
        raise NotImplementedError('not implemented')
    def list(self, limit: int) -> List[Product]:
        raise NotImplementedError('not implemented')
    def delete(self, id: str) -> None:
        raise NotImplementedError('not implemented')

File: benchmark/results/ts-monolith-sonnet-inline/run-005/b-baseline/output/src/catalog/services/ProductService.ts
import { Id } from '../../shared/value-objects/Id';
import { Product } from '../aggregates/Product';

/** Manages product lifecycle in the catalog. */
export class ProductService {
  create(input: Partial<Product>): Product { throw new Error('not implemented'); }
  findById(id: Id): Product | null { throw new Error('not implemented'); }
  list(limit: number): Product[] { throw new Error('not implemented'); }
  delete(id: Id): void { throw new Error('not implemented'); }
}
